From 54928d643ec0d201bf0101d67437b53d7fdbafad Mon Sep 17 00:00:00 2001
From: Karmjit Mahil <karmjit.mahil@igalia.com>
Date: Thu, 6 Feb 2025 10:34:49 +0100
Subject: [PATCH] loader/wayland: Fix missing timespec.h include

`loader_wayland_dispatch()` also makes use of `timespec` so we
need `timespec.h`. Otherwise it fails to build due to
`timespec_sub_saturate()` missing.

Signed-off-by: Karmjit Mahil <karmjit.mahil@igalia.com>
Fixes: 90effcceab7 ("wsi/wayland: refactor wayland dispatch")
Closes: https://gitlab.freedesktop.org/mesa/mesa/-/issues/12580
Reviewed-by: Eric Engestrom <None>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/33423>
---
 src/loader/loader_wayland_helper.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/loader/loader_wayland_helper.h b/src/loader/loader_wayland_helper.h
index 45f412a6c7ba1..0d7c49a15f246 100644
--- a/src/loader/loader_wayland_helper.h
+++ b/src/loader/loader_wayland_helper.h
@@ -23,11 +23,10 @@
 #ifndef LOADER_WAYLAND_HELPER_HEADER_H
 #define LOADER_WAYLAND_HELPER_HEADER_H
 
+#include <util/timespec.h>
 #include <wayland-client.h>
 
 #ifndef HAVE_WL_DISPATCH_QUEUE_TIMEOUT
-
-#include <util/timespec.h>
 int
 wl_display_dispatch_queue_timeout(struct wl_display *display,
                                   struct wl_event_queue *queue,
-- 
GitLab

